dashboard/test/controllers/omniauth_callbacks_controller_test.rb

```diff
@@ -451,6 +451,22 @@ def setup_should_connect_provider(user, timestamp = Time.now)
     end
   end
 
+  def assert_oauth_auth_option(user, oauth_hash)
+    assert_equal 1, user.authentication_options.count
+    auth_option = user.authentication_options.last
+
+    assert_equal user, auth_option.user
+    assert_equal User.hash_email(oauth_hash.info.email), auth_option.hashed_email
+    assert_equal oauth_hash.provider, auth_option.credential_type
+    assert_equal user.uid, auth_option.authentication_id
+    expected_auth_data = {
+      oauth_token: oauth_hash.credentials.token,
+      oauth_token_expiration: oauth_hash.credentials.expires_at,
+      oauth_refresh_token: oauth_hash.credentials.refresh_token
+    }
+    assert_equal expected_auth_data.to_json, auth_option.data
```

Comment:
Maybe we should extract and reuse the `assert_user` helper here.
https://github.com/code-dot-org/code-dot-org/blob/4a74fccce4104817ee67bc1d3c89cb5ba4de2777/dashboard/test/lib/user_multi_auth_helper_test.rb#L289-L297

Reply:
since i'm comparing a bunch of properties on `authentication_option`, so any opposition to me extracting `assert_authentication_option` and reusing that instead?

Reply:
Sounds great!